stdlib/http/src/main/java/org/ballerinalang/net/http/ClientHandshakeListener.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * The handshake listener for the client.
+ *
+ * @since 1.0.0
+ */
+public class ClientHandshakeListener extends WebSocketClientHandshakeListener {
+    private final WebSocketService wsService;
+    private final ClientListener clientConnectorListener;
+    private final boolean readyOnConnect;
+    private final ObjectValue webSocketClient;
+    private CountDownLatch countDownLatch;
+    private static final Logger logger = LoggerFactory.getLogger(ClientHandshakeListener.class);
+
+    public ClientHandshakeListener(ObjectValue webSocketClient, WebSocketService wsService,
+                                   ClientListener clientConnectorListener, boolean readyOnConnect,
+                                   CountDownLatch countDownLatch) {
+        super(webSocketClient, wsService, clientConnectorListener, readyOnConnect, countDownLatch);
+        this.webSocketClient = webSocketClient;
+        this.wsService = wsService;
+        this.clientConnectorListener = clientConnectorListener;
+        this.readyOnConnect = readyOnConnect;
+        this.countDownLatch = countDownLatch;
+    }
+
+    @Override
+    public void onSuccess(WebSocketConnection webSocketConnection, HttpCarbonResponse carbonResponse) {
+        //Using only one service endpoint in the client as there can be only one connection.
+        webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD, HttpUtil.createResponseStruct(carbonResponse));
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_PACKAGE_HTTP,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
+        WebSocketOpenConnectionInfo connectionInfo = new WebSocketOpenConnectionInfo(
+                wsService, webSocketConnection, webSocketClient);
+        webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
+        clientConnectorListener.setConnectionInfo(connectionInfo);
+        webSocketClient.set(WebSocketConstants.CLIENT_CONNECTOR_FIELD, webSocketConnector);
+        if (hasRetryConfig(webSocketClient)) {
+            if (readyOnConnect) {
+                webSocketConnection.readNextFrame();
+            }
+            WebSocketUtil.populateEndpoint(webSocketConnection, webSocketClient);
+            logger.info(CONNECTED_TO + webSocketClient.getStringValue(WebSocketConstants.
+                    CLIENT_URL_CONFIG));
+        } else {
+            RetryContext retryConfig = (RetryContext) webSocketClient.getNativeData(RETRY_CONFIG);
+            logger.info(CONNECTED_TO + webSocketClient.getStringValue(WebSocketConstants.
+                    CLIENT_URL_CONFIG));
+            setWebSocketEndpoint(retryConfig, webSocketClient, webSocketConnection);
+            if (retryConfig.isConnectionMade() || readyOnConnect) {
+                webSocketConnection.readNextFrame();
+            }
+            setReconnectContexValue(retryConfig);
```

Comment:
```suggestion
            setReconnectContextValue(retryConfig);
```
Same comment as above applies here as well.